adagrams/game.py

```diff
@@ -1,13 +1,122 @@
 from random import randint
 
+ALPHABET = [
+        "A", "B", "C", "D", "E", "F", "G",
+        "H", "I", "J", "K", "L", "M", "N",
+        "O", "P", "Q", "R", "S", "T", "U",
+        "V", "W", "X", "Y", "Z"
+]
+
 def draw_letters():
-    pass
+    ten_strings = []
+    # Pool dictionary to keep count of letters
+    LETTER_POOL = {
+        "A": 9, "B": 2, "C": 2, "D": 4, "E": 12, "F": 2, "G": 3,
+        "H": 2, "I": 9, "J": 1, "K": 1, "L": 4, "M": 2, "N": 6,
+        "O": 8, "P": 2, "Q": 1, "R": 6, "S": 4, "T": 6, "U": 4,
+        "V": 2, "W": 2, "X": 1, "Y": 2, "Z": 1
+    }
+
+    # While ten_strings doesn't have 10 letters -> execute loop
+    while len(ten_strings) < 10:
+        num = randint(0, 25)
+        letter = ALPHABET[num]
+        available = LETTER_POOL[letter]
+        
+        print(f"Letter: {letter} Pool amount: {available}")
+
+        # If the pool has letters available -> add to ten_strings
+        if available > 0:
+            LETTER_POOL[letter] = available - 1
+            ten_strings.append(letter)
+    
+    print(f"Ten Strings: {ten_strings}")
+    return ten_strings
 
-def uses_available_letters(word, letter_bank):
-    pass
+def uses_available_letters(word, letters):
+    letter_pool = get_letter_pool(letters)
+    
+    # Check each letter in the word
+    for letter in word.upper():  # Convert to uppercase
+        if letter in letter_pool and letter_pool[letter] > 0:
+            letter_pool[letter] -= 1
+        else:
+            return False
+    
+    return True
+
+# Helper function to create letter pool dictionary
+def get_letter_pool(letter_list):
+    count_dict = {}
+    for letter in letter_list:
+        if letter in count_dict:
+            count_dict[letter] += 1
+        else:
+            count_dict[letter] = 1
+    return count_dict
 
 def score_word(word):
-    pass
+    # Letter to score mapping
+    letter_scores = {
+        'A': 1, 'E': 1, 'I': 1, 'O': 1, 'U': 1, 'L': 1, 'N': 1, 'R': 1, 'S': 1, 'T': 1,
+        'D': 2, 'G': 2,
+        'B': 3, 'C': 3, 'M': 3, 'P': 3,
+        'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
+        'K': 5,
+        'J': 8, 'X': 8,
+        'Q': 10, 'Z': 10
+    }
+    
+    total_score = 0
+    
+    # If length of the word is 7, 8, 9, or 10, then the word gets an additional 8 points
+    if len(word) >= 7:
+        total_score += 8
```

Comment:
In this case, the number 8 is once again acting as a magic number that appeared without context. We could describe it by calling it something like what we have below:

```suggestion
        BONUS_POINTS = 8
        total_score += BONUS_POINTS
```